### PR TITLE
Feature/sie 77 account edit

### PR DESCRIPTION
**Pull request by cmutuc25**
This implements account edit and fixes some of the routing issues.

`Dashboard.js` now takes children to render the specific pages
`App.js` was changed to include all the routes and private routes
`PrivateRoute.js` way to encapsulate private routes

### CODE REVIEW

File: src/firebase/api/users.js [hunk 1]
```diff
@@ -134,6 +135,7 @@ export const generateUserDoc = async (userAuth, userData) => {
  * @return {User} user document
  */
 export const getUser = async (uid) => {
+  console.log('getUser');
```
Comment (alecherryy) [inline src/firebase/api/users.js:138]: @cmutuc25 found a lingering `console.log()` statement!

File: src/stories/pages/AccountPage/AccountPage.js [hunk 2]
```diff
@@ -19,46 +17,27 @@ export const UserContext = React.createContext();
...
+      ></HorizontalTitle>
+      <AccountInfoPage></AccountInfoPage>
+      <Button
+        className="account-page__delete"
+        modifierClasses="button--small"
+        isButton={true}
+        text="Delete My Account"
+      ></Button>
+      <Modal buttonText="Reset Password">
+        <ResetPasswordForm></ResetPasswordForm>
```
Comment (alecherryy) [inline src/stories/pages/AccountPage/AccountPage.js:39]: Should replace this with `<ResetPasswordForm />`

**Review by alecherryy — CHANGES_REQUESTED** [review]
Getting an error that prevents me from viewing the account page, per our Slack discussion.

File: src/stories/pages/AccountPage/AccountInfoPage/AccountInfoPage.js [hunk 3]
```diff
@@ -18,82 +21,95 @@ import HorizontalRuleLight from '../../../../images/icon-horizontal-rule-light.s
...
+  };
 
   return (
     <div className="account-info">
       <div className="account-info__section-header">
         <img
-          className="personal__horizontal-rule"
+          className="account-info__personal__horizontal-rule"
           src={HorizontalRuleDark}
         ></img>
```
Comment (alecherryy) [inline src/stories/pages/AccountPage/AccountInfoPage/AccountInfoPage.js:68]: Image tags should not have a closing tag; it's ``

File: src/stories/pages/AccountPage/AccountInfoPage/AccountInfoPage.js [hunk 4]
```diff
@@ -18,82 +21,95 @@ import HorizontalRuleLight from '../../../../images/icon-horizontal-rule-light.s
...
         <h5 className="account-info__header-text">PERSONAL INFO</h5>
-        <img className="account-info__edit" src={Edit}></img>
+        <img
+          className="account-info__edit"
+          src={Edit}
+          onClick={() => updateUserInfo()}
+        ></img>
```
Comment (alecherryy) [inline src/stories/pages/AccountPage/AccountInfoPage/AccountInfoPage.js:74]: Same as above.

**Review by fernandowinfield — COMMENTED** [review]
Two things to revise:

1.  When I `npm start` this branch locally the app crashes immediately with the following error:

`Error: Objects are not valid as a React child (found: object with keys {children}). If you meant to render a collection of children, use an array instead.`

I don't know if it's just me but might be worth checking in someone else's computer to see if it replicates or not. If it is an actual issue, the problem might be that either `children` needs to be mapped (to avoid rendering a JavaScript array object), or maybe `children` can be passed down as a prop.

2. Left some nit-pick comments about 'eslint' comments and a typo.

**Comment by alecherryy** [comment]
> Two things to revise:
> 
>     1. When I `npm start` this branch locally the app crashes immediately with the following error:
> 
> 
> `Error: Objects are not valid as a React child (found: object with keys {children}). If you meant to render a collection of children, use an array instead.`
> 
> I don't know if it's just me but might be worth checking in someone else's computer to see if it replicates or not. If it is an actual issue, the problem might be that either `children` needs to be mapped (to avoid rendering a JavaScript array object), or maybe `children` can be passed down as a prop.
> 
>     1. Left some nit-pick comments about 'eslint' comments and a typo.

Which component threw this error? There might me be some out of date code in here.

**Review by fernandowinfield — APPROVED** [review]

**Comment by cmutuc25** [comment]
Closing as this is included in https://github.com/neu-self-image-experiments/sie-firebase/pull/68

**Pull request closed without merging**